src/economy/economy.cpp

```diff
@@ -3424,193 +5521,349 @@ void daily_update(sys::state& state, bool initiate_buildings) {
+		auto states = state.world.market_get_zone_from_local_market(ids);
+		auto nations = state.world.state_instance_get_nation_from_state_ownership(states);
+
+		uint32_t total_commodities = state.world.commodity_size();
+		state.world.for_each_pop_type([&](dcon::pop_type_id pt) {
+			ve::fp_vector ln_total = 0.0f;
+			ve::fp_vector en_total = 0.0f;
+			ve::fp_vector lx_total = 0.0f;
+
+			ve::fp_vector ln_max = 0.0f;
+			ve::fp_vector en_max = 0.0f;
+			ve::fp_vector lx_max = 0.0f;
+
+			for(uint32_t i = 1; i < total_commodities; ++i) {
+				dcon::commodity_id c{ dcon::commodity_id::value_base_t(i) };
+				auto sat = state.world.market_get_demand_satisfaction(ids, c);
+				auto valid_mask = valid_need(state, nations, c);
+
+				auto ln_val = ve::select(valid_mask,
+					state.world.pop_type_get_life_needs(pt, c)
+					* state.world.market_get_life_needs_weights(ids, c), 0.f);
+
+				ln_total = ln_total + ln_val;
+				ln_max = ln_max + ln_val * sat;
+
+				auto en_val = ve::select(valid_mask,
+					state.world.pop_type_get_everyday_needs(pt, c)
+					* state.world.market_get_everyday_needs_weights(ids, c), 0.f);
+
+				en_total = en_total + en_val;
+				en_max = en_max + en_val * sat;
+
+				auto lx_val = ve::select(valid_mask,
+					state.world.pop_type_get_luxury_needs(pt, c)
+					* state.world.market_get_luxury_needs_weights(ids, c), 0.f);
+
+				lx_total = lx_total + lx_val;
+				lx_max = lx_max + lx_val * sat;
+			}
+
+			ln_max = ve::select(ln_total > 0.f, ln_max / ln_total, 1.f);
+			en_max = ve::select(en_total > 0.f, en_max / en_total, 1.f);
+			lx_max = ve::select(lx_total > 0.f, lx_max / lx_total, 1.f);
+
+			ve::apply([](float life, float everyday, float luxury) {
+				assert(life >= 0.f && life <= 1.f);
+				assert(everyday >= 0.f && everyday <= 1.f);
+				assert(luxury >= 0.f && luxury <= 1.f);
+			}, ln_max, en_max, lx_max);
+
+			state.world.market_set_max_life_needs_satisfaction(ids, pt, ln_max);
+			state.world.market_set_max_everyday_needs_satisfaction(ids, pt, en_max);
+			state.world.market_set_max_luxury_needs_satisfaction(ids, pt, lx_max);
+		});
+	});
+
+	sanity_check(state);
 
-		float factory_min_wage = pop_factory_min_wage(state, n, min_wage_factor);
-		float farmer_min_wage = pop_farmer_min_wage(state, n, min_wage_factor);
-		float laborer_min_wage = pop_laborer_min_wage(state, n, min_wage_factor);
+	// finally we can move to production:
+	// reset supply:
 
-		update_national_artisan_production(state, n);
+	state.world.for_each_commodity([&](dcon::commodity_id c) {
+		state.world.execute_serial_over_market([&](auto markets) {
+			state.world.market_set_supply(markets, c, ve::fp_vector{});
+			state.world.market_set_import(markets, c, ve::fp_vector{});
+			state.world.market_set_export(markets, c, ve::fp_vector{});
+		});
+	});
+
+	// we can handle each trade good separately: they do not influence each other
+	// 
+	// register trade supply
+	concurrency::parallel_for(uint32_t(0), total_commodities, [&](uint32_t k) {
```

Comment:
Assuming you don't allow the money commodity to be traded, you can start this parallel loop at `uint32_t(1)`, since it is always id 0 (mods that add additional such commodities mean that the check for the money commodity is still necessary of course)